test/e2e/dns_ingressdegrade_test.go

```diff
@@ -86,20 +86,22 @@ func updateDNSConfig(set bool, dnsConfig *configv1.DNS) {
 	// Tested on Azure, AWS, and GCP.  Azure privateZone uses "/" notation,
 	// so prepending "error-" without "/" does not cause a failure on Azure.
 	const injectError = "/error"
-	if dnsConfig.Spec.PrivateZone.ID != "" {
-		if set {
-			dnsConfig.Spec.PrivateZone.ID = injectError + dnsConfig.Spec.PrivateZone.ID
-		} else {
-			// Remove injectError from prefix.
-			dnsConfig.Spec.PrivateZone.ID = dnsConfig.Spec.PrivateZone.ID[len(injectError):]
+	if dnsConfig.Spec.PrivateZone != nil {
```

Comment:
This will cause step 3 of `TestIngressStatus` to fail, right?  `TestIngressStatus` needs a skip when the private zone is not specified.  

It would be even better if we had tests for both the private zone and public zone, with an appropriate skip for each zone if it isn't set.  

Historical note: #641 added the original test.  At that point, I believe, putting an invalid value in the private zone would cause the clusteroperator to report a problem, but putting an invalid value in the public zone would not.  #967 subsequently modified the status reporting so that an invalid value in either public or private zone config would cause the clusteroperator to report an error.  In retrospect, we should have updated this test as part of #967.

Reply:
Before skipping this test I want to know what scenarios we want to test here.
Do we want to test:

- Public empty
  Private set

- Public set
  Private set

- Both set

- Both empty (I guess that's not possible?)

Reply:
I think all combinations are possible:

- As far as I know, most cloud platforms have both public and private set.
- Today I learned that OpenStack has public set and private nil.  
- An internal/private cluster can have public nil and private set.
- Public and private can both be nil on, say, a bare-metal cluster; the cluster-admin must configure DNS manually.  

The simplest thing would be to split `TestIngressStatus` into two tests: one for the public zone, and one for the private zone, each with a skip rule if the zone is not specified.  Most of the current test logic can be factored into new helper function, so each test can be just a skip rule and then a call to the helper.

Reply:
I've updated it. Let me know if you like the logic, or if you really want it split by tests that are skipped when needed.